packages/common/src/services/auth/hedgehog.ts

```diff
@@ -175,5 +185,42 @@ export const createHedgehog = ({
     }
   }
 
+  const originalConfirmCredentials = hedgehog.confirmCredentials.bind(hedgehog)
+  hedgehog.confirmCredentials = async ({
+    softCheck,
+    ...confirmCredentialsArgs
+  }: ConfirmCredentialsArgs) => {
+    if (softCheck) {
+      const lookupKey = await getLookupKey({
+        username: confirmCredentialsArgs.username,
+        password: confirmCredentialsArgs.password
+      })
+      try {
+        const wallet = hedgehog.wallet
+        if (!wallet) {
+          throw new Error(
+            'Cannot check credentials - user is not authenticated'
+          )
+        }
+        const headers = await getAuthHeaders(wallet)
+        if (headers[AuthHeaders.Message] && headers[AuthHeaders.Signature]) {
+          return await makeIdentityRequest({
+            url: '/authentication/check',
+            method: 'GET',
+            headers,
+            params: { lookupKey }
+          })
+        } else {
+          throw new Error(
+            'Cannot check credentials - user is not authenticated'
+          )
+        }
+      } catch (e) {
+        return false
+      }
+    }
+    return await originalConfirmCredentials(confirmCredentialsArgs)
```

Comment:
The original implementation of this pattern in libs wrapped a function around Hedgehog that added this `softCheck` mode and then would fall back to Hedgehog to do the full check otherwise.
The pattern in this class at the moment (and it should be changed later) is to pull the original method off the Hedgehog instance, then patch it with a version that supports some extended parameters and behavior.
I think the goal was to add extra behavior to Hedghog without needing to duplicate all the methods with their types and implement them as wrappers that just call a Hedghog instance.

Ultimately, I think that pattern will be cleaner and easier to follow.